File: backend/src/main/java/com/backend/global/openai/OpenAIKeyMissingCondition.java
package com.backend.global.openai;

import org.springframework.context.annotation.Condition;
import org.springframework.context.annotation.ConditionContext;
import org.springframework.core.type.AnnotatedTypeMetadata;

public class OpenAIKeyMissingCondition implements Condition {
    @Override
    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
        return OpenAIKeyUtil.findKey() == null;
    }
}

File: backend/src/main/java/com/backend/global/openai/OpenAIKeyMissingCondition.kt
package com.backend.global.openai

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.core.type.AnnotatedTypeMetadata

class OpenAIKeyMissingCondition : Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        return OpenAIKeyUtil.findKey() == null
    }
}
